### ISSUE TITLE
JSON Schema: items cannot be added to arrays within multischema fields

### ISSUE THREAD

**Issue by DallasHoff** [ngx-formly/ngx-formly#3769]
**Description**

When a "multischema" field created from a oneOf array in a JSON schema contains an array field, it is not possible to add items to that array field. When the `add` method of `FieldArrayType` is called, an error is thrown: `this.model.splice is not a function`, because `model` is `undefined`. The field's model is not initialized with an array as other array fields are with other form configurations, but the `add` method assumes that it is.

I attempted to override the `add` method in a class that extends `FieldArrayType` and add a check that makes sure `model` is an array, but I found that that is not possible because the `model` property is readonly.

**Minimal Reproduction**

I have modified the oneOf form in the Formly JSON schema example to demonstrate this. The error can be observed by switching the dropdown to the "ipsum" option and clicking the plus button.

https://stackblitz.com/edit/angular-zzhwis?devToolsHeight=33&file=src%2Fassets%2Fjson-schema%2FoneOf.json

```json
{
  "schema": {
    "title": "Lorem or Ipsum",
    "oneOf": [
      {
        "title": "lorem",
        "type": "string",
        "default": ""
      },
      {
        "title": "ipsum",
        "type": "array",
        "items": {
          "type": "string",
          "default": ""
        },
        "default": [""]
      }
    ]
  },
  "model": {}
}
```

**Your Environment**

- Angular version: 16.1.4
- Formly version: 6.1.8

**Comment by DallasHoff** [comment]
In my use case, the `oneOf` is inside the `items` of an array field, so I am trying to use the schema to represent an array where the items in the array can be strings or arrays. Putting the `oneOf` inside an object would be a change to the shape of the data. Is there no way to have Formly represent the shape of data that I described? Ideally, the shape of the data should not matter to Formly, and it would be able to represent any valid schema.

**Comment by kenisteward** [comment]
Currently formly supports one of structures of similar types. 

When processing schemas the parawr has to reason about a type at runtime but before rendering in order to choose the correct model type / control type / formControl to have data applied.

One of the restrictions of this is because the way that model data gets added to form controls.  Via the angular reactive API arrays are treated with the formArray implementation, objects with formGroup and primitives with formControl.  This means when trying to validate on a one of that has more than one base type the engine would have to create a separate form controls type of each of these types of data.  While not completely impossible it is definitely a non trivial feature to add and it hasn't been gotten to yet.

I've had this same type of requirement in forms I've made but it turned out the way the data was being modeled baseline was better by not using this strategy. Is there a particular reason why you need 1 field to represent multiple disparate types?

**Comment by DallasHoff** [comment]
> Is there a particular reason why you need 1 field to represent multiple disparate types instead of having it be an array of objects with disparate keys so the base array can be treated as a single internal type?

Honestly no, besides the fact that it would require making a change to our JSON schema and accordingly updating the other services that use that schema. Like I said before, Formly would **ideally** handle any valid schema, but I can see what you are saying about how this would be difficult, and I think we should be able to work around this restriction. Thanks for your help.

**Comment by kenisteward** [comment]
No problem glad I could help! Hopefully in the future we will be able to
provide this not just in object types.

**Comment by rishabh413** [comment]
Hi @kenisteward 

Thanks for the explanation. Please guide us on https://github.com/ngx-formly/ngx-formly/issues/3723 as well, if possible. 

Is the reasoning on https://github.com/ngx-formly/ngx-formly/issues/3769#issuecomment-1714581285 related to  https://github.com/ngx-formly/ngx-formly/issues/3723 as well?

**Comment by kenisteward** [comment]
> Hi @kenisteward 
> 
> Thanks for the explanation. Please guide us on https://github.com/ngx-formly/ngx-formly/issues/3723 as well, if possible. 
> 
> Is the reasoning on https://github.com/ngx-formly/ngx-formly/issues/3769#issuecomment-1714581285 related to  https://github.com/ngx-formly/ngx-formly/issues/3723 as well?

I'll try to remember to give a fuller explanation on this when I'm at my computer tomorrow but yes it falls on this order.

I had explicitly looked at this ticket a while back and made wrappers that fixed some of those internal workings for our projects at my job. I however had a data structure that allowed me to make it easier for the formly engine to pick the correct schema via a type key that had a const of the properly selected schema

**Comment by aitboudad** [comment]
This issue has been fixed and released as part of [v6.2.0](https://github.com/ngx-formly/ngx-formly/releases/tag/v6.2.0) release.

Please let us know, in case you are still encountering a similar issue/problem.
Thank you!
